pynicotine/gtkgui/dialogs/download.py

```diff
@@ -0,0 +1,594 @@
+            self.tree_view.set_row_value(iterator, "selected", selected)
+
+            if not selected:
+                continue
+
+            row_id = self.tree_view.get_row_value(iterator, "id_data")
+
+            if row_id not in self.parent_iterators:
+                self.num_selected_files += 1
+                self.total_selected_size += self.tree_view.get_row_value(iterator, "size_data")
+
+        self.unselect_all_button.set_visible(False)
+        self.update_title()
+
+    def on_expand_tree(self, *_args):
+
+        if not self.expand_button.get_visible():
+            return
+
+        expanded = self.expand_button.get_active()
+
+        if expanded:
+            icon_name = "view-restore-symbolic"
+            self.tree_view.expand_all_rows()
+        else:
+            icon_name = "view-fullscreen-symbolic"
+            self.tree_view.collapse_all_rows()
+
+        icon_args = (Gtk.IconSize.BUTTON,) if GTK_API_VERSION == 3 else ()  # pylint: disable=no-member
+        self.expand_icon.set_from_icon_name(icon_name, *icon_args)
+
+    def on_select_file(self, tree_view, iterator):
+
+        selected = not tree_view.get_row_value(iterator, "selected")
+        username = tree_view.get_row_value(iterator, "user_data")
+        folder_path = tree_view.get_row_value(iterator, "folder_path_data")
+        row_id = tree_view.get_row_value(iterator, "id_data")
+
+        tree_view.set_row_value(iterator, "selected", selected)
+
+        if row_id in self.parent_iterators:
+            for i_iterator in tree_view.iterators.values():
+                i_folder_path = tree_view.get_row_value(i_iterator, "folder_path_data")
+
+                if i_folder_path != folder_path:
+                    continue
+
+                i_selected = tree_view.get_row_value(i_iterator, "selected")
+
+                if selected and not i_selected:
+                    self.num_selected_files += 1
+                    self.total_selected_size += self.tree_view.get_row_value(i_iterator, "size_data")
+
+                elif not selected and i_selected:
+                    self.num_selected_files -= 1
+                    self.total_selected_size -= self.tree_view.get_row_value(i_iterator, "size_data")
+
+                tree_view.set_row_value(i_iterator, "selected", selected)
+        else:
+            parent_iterator = self.parent_iterators[username + folder_path]
+
+            if not selected:
+                tree_view.set_row_value(parent_iterator, "selected", selected)
+
+            if selected:
+                self.num_selected_files += 1
+                self.total_selected_size += tree_view.get_row_value(iterator, "size_data")
+            else:
+                self.num_selected_files -= 1
+                self.total_selected_size -= tree_view.get_row_value(iterator, "size_data")
+
+        self.update_title()
+
+    def on_download_folder_selected(self):
+        self.has_custom_download_location = True
```

Comment:
This might not be `True` if the user happens to manually select their default download folder. This would lead to a bug where the "usernamesubfolders" setting is not respected in `on_download()`.